client/components/RunningGraph.vue

```diff
@@ -33,61 +37,72 @@
  import { eventBus } from '../client';
  import { getBackgroundColorFor, getBorderColorFor, lineWidthInPixels, pointRadiusInPixels} from '../client'; 
  import { truncate} from '../pages/utilities.js'; 
+ import { isPathIn} from '../pages/utilities.js'; 
 
   export default {
     components: {
       LineChart,
     },
     props: [
+      'routes',
       'stats',
-      'statshistory'
+      'statshistory',
+      'appname'
     ],
     data () {
       return {
         datacollection: null,
+        total: 0
       }
     },
     mounted () {
       //this.updateChart();
     },
     methods: {
       updateChart () {
+        // update the chart to display data for the routes in "routes", or for all routes if "routes" is not set 
+        var totalCount = 0; 
         if (this.statshistory && this.stats){
           this.datacollection = {};
           this.datacollection["labels"]= this.statshistory.map(eachStatistic => "" );
           this.datacollection["datasets"]=[];
           for (var thisPath in this.stats.FunctionStatsMap){
-            var dataSetForPath = {
-              label: thisPath + ": "+ this.stats.FunctionStatsMap[thisPath].Running,
-              fill: false, // Only use fill if chart is stacked
-              backgroundColor: 'white', // needed if fill is false to set fill color in legend
-              borderColor: getBorderColorFor(thisPath),
-              borderWidth: lineWidthInPixels,
-              radius:pointRadiusInPixels,
-              data: this.statshistory.map(eachStatistic => eachStatistic.FunctionStatsMap[thisPath].Running)
-            };
-            this.datacollection["datasets"].push(dataSetForPath);
-             
+            if (this.routes==null || isPathIn(thisPath,this.routes)){
```

Comment:
Common code see https://github.com/fnproject/ui/pull/37/files#diff-a3db016c87a936e401dcbc268b43db7cR70

Reply:
Done. All four graphs now use a shared function in `graphUtilities.js`.